server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/ZRANGE.java

```diff
@@ -3,49 +3,262 @@
 import io.netty.channel.ChannelHandlerContext;
 import org.infinispan.multimap.impl.EmbeddedMultimapSortedSetCache;
 import org.infinispan.multimap.impl.SortedSetBucket;
+import org.infinispan.multimap.impl.SortedSetSubsetArgs;
 import org.infinispan.server.resp.Consumers;
 import org.infinispan.server.resp.Resp3Handler;
 import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
 import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.ArgumentUtils;
 import org.infinispan.server.resp.commands.Resp3Command;
 
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
 import java.util.concurrent.CompletionStage;
 
 /**
- * TODO: implement all the command, this is just the start point to help testing ZADD
+ * Returns the specified range of elements in the sorted set stored at <key>.
+ * <p>
+ * ZRANGE can perform different types of range queries: by index (rank), by the score,or by lexicographical order.
+ * <p>
+ * Starting with Redis 6.2.0, this command can replace the following commands:
+ * <ul>
+ *    <li>{@link ZREVRANGE}/li>
+ *    <li>{@link ZRANGEBYSCORE}</li>
+ *    <li>{@link ZREVRANGEBYSCORE}</li>
+ *    <li>{@link ZRANGEBYLEX}</li>
+ *    <li>{@link ZREVRANGEBYLEX}</li>
+ * </ul>
  *
- * @since 15.0
  * @see <a href="https://redis.io/commands/zrange">Redis Documentation</a>
+ * @since 15.0
  */
 public class ZRANGE extends RespCommand implements Resp3Command {
+   protected enum Arg {
+      BYSCORE, BYLEX, REV, LIMIT, WITHSCORES
+   }
+
+   private boolean isRev;
+   private boolean byScore;
+   private boolean byLex;
```

Comment:
I think using these variables as instance variables might be causing an issue. The ZRANGE is shared between all client connections, so we need to synchronize the access. And we are not resetting after execution, so mixing the flags raise issues:

Here is one client in one terminal:

```text
127.0.0.1:11222> zrange my-set 0 3 BYSCORE
1) "a"
2) "b"
3) "c"
127.0.0.1:11222> zrange my-set a c BYLEX
(error) ERR syntax error
127.0.0.1:11222> zrange my-set 0 3 BYSCORE
(error) ERR syntax error
```

And another client in another terminal, freshly connected:

```text
127.0.0.1:11222> auth admin password
OK
127.0.0.1:11222> zrange my-set 0 3
(error) ERR syntax error
```

Reply:
yes, I realized this and solved this in ZRANGESTORE